### PR TITLE
makes it so that students allways show up in csv

### PR DESCRIPTION
**Pull request by dittonjs**

### CODE REVIEW

**Comment by BenHSmith13** [comment]
I think this looks good, but @nick-benoit14 should probably once over it too

File: app/views/exports/attendances.csv.erb [hunk 1]
```diff
@@ -1,21 +1,19 @@
...
-<% att_by_student = @attendances.group_by{|a| a.sortable_name} %>
-<% days =  att_by_date.keys.sort %>
+<% days = @attendances.group_by{|a| a.date}.keys.sort %>
 <%- header_row = ['Name'].concat days -%>
 <%= CSV.generate_line(header_row, {:row_sep => ''}) %>
-<%- att_by_student.keys.sort.each do |key|
-  student_att = att_by_student[key]
-  if att_by_student[key][0][:name].include?(',')
-    parts = att_by_student[key][0][:name].split(',')
+<%- sorted_students = @students.sort_by{|s| s["sortable_name"]} -%>
```
Comment (bfcoder) [inline app/views/exports/attendances.csv.erb]: Any reason all this logic can't go into the controller/model? Logic like this deserves no place in the erb..

**Comment by dittonjs** [comment]
or cory

File: app/views/exports/attendances.csv.erb (hunk 1 above)
Comment (dittonjs) [inline app/views/exports/attendances.csv.erb]: i can do that i think. I should just be able to move this logic the exact same way it is to the controller right @bfcoder?

**Comment by bfcoder** [comment]
This looks much better. Now your controller is kinda fat, consider moving a lot of the code into its own class?

**Comment by BenHSmith13** [comment]
This could use a test

**Comment by BenHSmith13** [comment]
this looks good to me

File: app/controllers/exports_controller.rb [hunk 2]
```diff
@@ -1,30 +1,29 @@
...
+    final_csv = AttendanceExportsHelper.generate_csv(students, attendances)
+    send_data(final_csv, filename: "attendance_export")
   end
 
   private
 
   def get_attendances
     attendances = Attendance.where(lms_course_id: params[:course_id])
-    if(params[:startDate] && params[:endDate])
+    if params[:startDate] && params[:endDate]
```
Comment (bfcoder) [inline app/controllers/exports_controller.rb]: use `.present?`

Comment (dittonjs) [inline app/controllers/exports_controller.rb]: i didnt write this but sure

Comment (bfcoder) [inline app/controllers/exports_controller.rb]: I know, but improve code when you can, right? :]

**Pull request merged**